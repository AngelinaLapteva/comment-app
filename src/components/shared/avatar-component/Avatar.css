.Avatar {
    width: auto;
    height: 40px;
    clip-path: circle(20px at center);
    margin-right: 15px;
  }