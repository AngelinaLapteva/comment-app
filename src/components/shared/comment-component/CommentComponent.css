
.commentWrapper {
    margin: 20px;
}
.commentWrapper:hover .hoverComment {
    color: #232324;
}
.iconXs {
    font-size: 2px;
    line-height: 25px;
}
