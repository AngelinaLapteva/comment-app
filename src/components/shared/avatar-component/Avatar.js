import React from 'react';
import './Avatar.css'

const avatar = props => {
    return (
        <img className="Avatar" alt="profile" src={props.avatarPath} />
    )
}
export default avatar;