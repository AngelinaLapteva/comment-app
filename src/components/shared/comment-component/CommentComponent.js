import React from 'react';
import Avatar from '../avatar-component/Avatar'
import './CommentComponent.css'
import { faChevronUp, faChevronDown, faCircle } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";


const commentComp = props => {
    const clickOnReply = () => {
        console.log('you clicked on reply')
    }
    return (props.comments.map(comment => {
        let replyText = ' reply'
        if (comment.amountOfReplies > 1) {
            replyText = ' replies'
        }
        return (
            <div key={comment.id} className="commentWrapper">
                <div className="rowView">
                    <Avatar avatarPath={comment.avatarPath} />
                    <div className="columnView">
                        <div className="rowView mb1">
                            <div className="darkGrey boldText">{comment.authorName}</div>
                            <div className="uppercaseText smallFont paddingXsm">
                                <span className="tagStyle">{comment.type}</span>
                            </div>
                            <div className="uppercaseText mediumGrey smallFont rowView centerAligned">
                                <FontAwesomeIcon className="iconXs paddingXsm " icon={faCircle} />
                                {comment.timeStamp}
                            </div>
                        </div>
                        <p className="marginY0 mb1">{comment.commentText}</p>
                        <div className="rowView">
                            <div className="uppercaseText lightGrey smallFont hoverComment pointer"
                                onClick={clickOnReply}>reply</div>
                            <div className="uppercaseText lightGrey smallFont paddingXRegular hoverComment pointer"
                                onClick={props.clicked}>
                                <span className="mediumGrey hoverComment ">
                                    {comment.amountOfReplies}
                                </span>
                                {replyText}
                            </div>
                            <div className="uppercaseText lightGrey smallFont paddingXRegular rowView centerAligned hoverComment pointer">
                                <FontAwesomeIcon className="regularFont paddingXsm" icon={faChevronUp} />
                                {comment.upCount}
                            </div>
                            <div className="uppercaseText lightGrey smallFont paddingXRegular rowView centerAligned hoverComment pointer">
                                <FontAwesomeIcon className="regularFont paddingXsm" icon={faChevronDown} />
                                {comment.downCount}
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    })
    )
}
export default commentComp;