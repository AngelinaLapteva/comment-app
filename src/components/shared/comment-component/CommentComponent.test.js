
import React from 'react';
import { render } from '@testing-library/react';
import CommentComponent from './CommentComponent';
import renderer from 'react-test-renderer';
import Avatar from '../avatar-component/Avatar'
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })
const comments = [
    {
        id: 'dji7492ds',
        authorName: 'TestName',
        type: 'author',
        timeStamp: '2 minutes ago',
        amountOfReplies: 21,
        upCount: 123,
        downCount: 0,
        avatarPath: '/img/catAvatr.png',
        commentText: 'So what the German automaker is likely to focus on today is the bigger picture. This will be the first time we see the Taycan free from prototype bodyworks.'
    }
]
const commentsTest2 = [
    {
        id: 'dji7492ds',
        authorName: 'TestName',
        type: 'author',
        timeStamp: '2 minutes ago',
        amountOfReplies: 1,
        upCount: 123,
        downCount: 0,
        avatarPath: '/img/catAvatr.png',
        commentText: 'So what the German automaker is likely to focus on today is the bigger picture. This will be the first time we see the Taycan free from prototype bodyworks.'
    }
]

describe(' Comment  component', () => {
    test('renders comment component with multiple replies', () => {

        const { getByText } = render(<CommentComponent comments={comments} />);
        const linkElement = getByText(/TestName/i);
        const replies = getByText(/replies/i);
        expect(linkElement).toBeInTheDocument();
        expect(replies).toBeInTheDocument();
    });

    test('renders comment component with one reply', () => {

        const { getAllByText } = render(<CommentComponent comments={commentsTest2} />);
        const replies = getAllByText(/reply/i);
        expect(replies).toHaveLength(2);
    });
    test('check the snapshot', () => {
        const component = renderer.create(
            <CommentComponent comments={comments} />
        );
        let commentCmp = component.toJSON();
        expect(commentCmp).toMatchSnapshot();
    })
    test('check if Avatar and FontAwesome component created', () => {
        const comWrap = shallow(<CommentComponent comments={comments}/>)
        expect(comWrap.find(Avatar)).toHaveLength(1);
        expect(comWrap.find(FontAwesomeIcon)).toHaveLength(3)
    })
})