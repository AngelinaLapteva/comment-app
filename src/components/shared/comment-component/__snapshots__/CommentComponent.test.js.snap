// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Comment  component check the snapshot 1`] = `
<div
  className="commentWrapper"
>
  <div
    className="rowView"
  >
    <img
      alt="profile"
      className="Avatar"
      src="/img/catAvatr.png"
    />
    <div
      className="columnView"
    >
      <div
        className="rowView mb1"
      >
        <div
          className="darkGrey boldText"
        >
          TestName
        </div>
        <div
          className="uppercaseText smallFont paddingXsm"
        >
          <span
            className="tagStyle"
          >
            author
          </span>
        </div>
        <div
          className="uppercaseText mediumGrey smallFont rowView centerAligned"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-circle fa-w-16 iconXs paddingXsm "
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          2 minutes ago
        </div>
      </div>
      <p
        className="marginY0 mb1"
      >
        So what the German automaker is likely to focus on today is the bigger picture. This will be the first time we see the Taycan free from prototype bodyworks.
      </p>
      <div
        className="rowView"
      >
        <div
          className="uppercaseText lightGrey smallFont hoverComment pointer"
          onClick={[Function]}
        >
          reply
        </div>
        <div
          className="uppercaseText lightGrey smallFont paddingXRegular hoverComment pointer"
        >
          <span
            className="mediumGrey hoverComment "
          >
            21
          </span>
           replies
        </div>
        <div
          className="uppercaseText lightGrey smallFont paddingXRegular rowView centerAligned hoverComment pointer"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-chevron-up fa-w-14 regularFont paddingXsm"
            data-icon="chevron-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          123
        </div>
        <div
          className="uppercaseText lightGrey smallFont paddingXRegular rowView centerAligned hoverComment pointer"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-chevron-down fa-w-14 regularFont paddingXsm"
            data-icon="chevron-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;
